lib/google_contacts_api/contacts_accessor.rb

```diff
@@ -25,5 +25,37 @@ def get_contacts(params = {})
       end
       GoogleContactsApi::ContactSet.new(response.body, @api)
     end
+
+    def post_contact(contact)
+      params = {}
+      headers = {
+          'Content-type' => 'application/json'
+          }
+      body = contact.json
+      params[:body] = body
+      url = "contacts/default/full"
+      response = @api.post_v2(url, params, headers)
+      GoogleContactsApi::Api.parse_response_code(response)
+    end
+
+    def put_contact(contact)
+      params = {}
+      headers = {
+          'Content-type' => 'application/json',
+          'If-Match' => '*'
+          }
+      body = contact.json
+      params[:body] = body
+      response = @api.put_v2(contact.edit_link, params, headers)
+      GoogleContactsApi::Api.parse_response_code(response)
+    end
+
+    def put_or_post_contact(contact)
```

Comment:
Yeah I get it now. I feel like the `save_contact` method in User is doing too little then. It's just wrapping some JSON in an object, which the gem's user should be able to do themselves. Maybe a structure like this would work?


    class API
      def post(contact);end
      def put(contact);end
      def exists?(contact);end
      def save_contact(contact)
        contact.exists? put(contact) : post(contact)
      end
    end


    > contact = GoogleContact.new(json)
    > API.save_contact(contact)
    => <#GoogleAPIResponse code=201 message="Created">